### PR TITLE
fix connection to ipv6 address + use fallback to jid domain

### LINKED ISSUE #73: Handle IPv6
Does not handle IPv6 correct

https://github.com/FluuxIO/go-xmpp/blob/7a386ec8d04c376aecd99d4cc109e934693005f6/client.go#L118-L119

### PR DESCRIPTION
**Pull request by genofire**
fixed #73

### CODE REVIEW

**Comment by mremond** [comment]
Thanks for the PR 👍 
I moved the code to add tests, and thus could find some cases where you need to add brackets when you append the port.
We should be good for even more cases.

File: network_test.go [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+		name  string
+		input string
+		want  string
+	}{
+		{name: "ipv4-no-port-1", input: "localhost", want: "localhost:5222"},
+		{name: "ipv4-with-port-1", input: "localhost:5555", want: "localhost:5555"},
+		{name: "ipv4-no-port-2", input: "127.0.0.1", want: "127.0.0.1:5222"},
+		{name: "ipv4-with-port-2", input: "127.0.0.1:5555", want: "127.0.0.1:5555"},
+		{name: "ipv6-no-port-1", input: "1::", want: "[1::]:5222"},
+		{name: "ipv6-no-port-2", input: "[1::]", want: "[1::]:5222"},
```
Comment (genofire) [inline network_test.go]: i prefer `::1` :stuck_out_tongue_winking_eye:

Comment (mremond) [reply network_test.go]: Ah, yes, I took some examples online but they are not very good. We should adapt them.

Comment (mremond) [reply network_test.go]: Examples are now a bit nicer: https://github.com/FluuxIO/go-xmpp/commit/f79a3a219bfae1368ee12e07e28a9f49c349a500

**Pull request merged**